src/ILLink.Tasks/CreateRuntimeRootDescriptorFile.cs

```diff
@@ -239,6 +290,70 @@ void OutputXml (string iLLinkTrimXmlFilePath, string outputFileName)
 			XmlDocument doc = new XmlDocument ();
 			doc.Load (iLLinkTrimXmlFilePath);
 			XmlNode linkerNode = doc["linker"];
+
+			if (featureSwitchMembers.Count > 0) {
+				foreach (var fsMembers in featureSwitchMembers.Keys) {
+					// <assembly fullname="System.Private.CoreLib" feature="System.Diagnostics.Tracing.EventSource.IsSupported" featurevalue="true" featuredefault="true">
+					XmlNode featureAssemblyNode = doc.CreateElement ("assembly");
+					XmlAttribute featureAssemblyFullName = doc.CreateAttribute ("fullname");
+					featureAssemblyFullName.Value = "System.Private.CoreLib";
+					featureAssemblyNode.Attributes.Append (featureAssemblyFullName);
+
+					XmlAttribute featureName = doc.CreateAttribute ("feature");
+					featureName.Value = fsMembers.feature;
+					featureAssemblyNode.Attributes.Append (featureName);
+
+					XmlAttribute featureValue = doc.CreateAttribute ("featurevalue");
+					featureValue.Value = fsMembers.featureValue;
+					featureAssemblyNode.Attributes.Append (featureValue);
+
+					XmlAttribute featureDefault = doc.CreateAttribute ("featuredefault");
+					featureDefault.Value = fsMembers.featureDefault;
+					featureAssemblyNode.Attributes.Append (featureDefault);
+
+					foreach (string typeName in featureSwitchMembers[fsMembers].Keys) {
+						XmlNode typeNode = doc.CreateElement ("type");
+						XmlAttribute typeFullName = doc.CreateAttribute ("fullname");
+						typeFullName.Value = typeName;
+						typeNode.Attributes.Append (typeFullName);
+
+						ClassMembers members = featureSwitchMembers[fsMembers][typeName];
+
+						if (members.keepAllFields) {
+							XmlAttribute preserve = doc.CreateAttribute ("preserve");
```

Comment:
Can we deduplicate the code here? I would expect the logic how to write ClassMembers to be identical with or without feature switch.